core/src/main/java/io/grpc/internal/StatsTraceContext.java

```diff
@@ -72,6 +73,7 @@
   private final Stopwatch stopwatch;
   private final Side side;
   private final Metadata.Key<StatsContext> statsHeader;
+  private volatile long clientPendingMillis = -1;
```

Comment:
millis are rather coarse, no?  I think it would be better if we used nanos for all time information throughout the library.